src/Tasks/WriteCodeFragment.cs

```diff
@@ -156,6 +156,10 @@ private string GenerateCode(out string extension)
                 Log.LogErrorWithCodeFromResources("WriteCodeFragment.CouldNotCreateProvider", Language, e.Message);
                 return null;
             }
+            finally
+            {
+                provider?.Dispose();
+            }
 
             extension = provider.FileExtension;
```

Comment:
`provider` is used here and then again further down. The new `Dispose()` call looks incorrect.

Reply:
I packed all provider-depended parts with try/catch